feat(sitemap): add namespaces configuration option (#14285)
withastro/astro: .changeset/rich-insects-scream.md

Reviewer: Let's add a code snippet with an example. 

Talk about an example where the user wants to remove the `video` namespace. We'll have a code snippet of the configuration, and then a code snippet of the resultant XML. It doesn't have to be comprehensive, just a piece where we show that the video namespace isn't there

---
'@astrojs/sitemap': minor
---

Adds a new configuration option `namespaces` for more control over XML namespaces used in sitemap generation

Excluding unused namespaces can help create cleaner, more focused sitemaps that are faster for search engines to parse and use less bandwidth. If your site doesn't have news content, videos, or multiple languages, you can exclude those namespaces to reduce XML bloat.

> The `namespaces` option allows you to configure `news`, `xhtml`, `image`, and `video` namespaces independently. All namespaces enabled by default for backward compatibility and no changes to existing projects is necessary. But now you can choose to streamline your XML and avoid unnecessary code.

Reviewer 2: ```suggestion
The `namespaces` option allows you to configure `news`, `xhtml`, `image`, and `video` namespaces independently. All namespaces are enabled by default for backward compatibility and no change to existing projects is necessary. But now, you can choose to streamline your XML and avoid unnecessary code.
```

---
'@astrojs/sitemap': minor
---

> Add configurable XML namespaces support to sitemap generation
> 
> - Add `namespaces` option to control which XML namespaces are included
> - Support for news, xhtml, image, and video namespaces
> - All namespaces enabled by default for backward compatibility

Reviewer 2: ```suggestion
Adds a new configuration option `namespaces` for more control over XML namespaces used in sitemap generation

Excluding unused namespaces can help create cleaner, more focused sitemaps that are faster for search engines to parse and use less bandwidth. If your site doesn't have news content, videos, or multiple languages, you can exclude those namespaces to reduce XML bloat.

The `namespaces` option allows you to configure `news`, `xhtml`, `image`, and `video` namespaces independently. All namespaces enabled by default for backward compatibility and no changes to existing projects is necessary. But now you can choose to streamline your XML and avoid unnecessary code.
```

Just tried to add some extra context here, pulling from what we ended up with in docs. For a changeset, giving people the extra context can be helpful for them to know why they should care about the new feature added! (And, we usually pull from here to write content for the release blog post, so it can be a bit more "hype" for your feature here!)

> Example: Removing the video namespace
> If you want to exclude the video namespace from your sitemap, set video: false in your configuration:

Reviewer 2: ```suggestion
For example, to exclude the video namespace from your sitemap, set `video: false` in your configuration:
```

```
// astro.config.mjs
import { sitemap } from '@astrojs/sitemap';

export default {
  integrations: [
    sitemap({
      namespaces: {
        video: false,
        // other namespaces remain enabled by default
      }
    })
  ]
};
```

> The generated XML will not include the xmlns:video namespace:

Reviewer 2: ```suggestion
The generated XML will not include the `xmlns:video` namespace:
```
I'm guessing this is the kind of thing we'd inline code for formatting. I LOVE that you provided the XML code block below! (See my other comment re: whether you think making a diff makes sense)

```
<?xml version="1.0" encoding="UTF-8"?>
<urlset
  xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
  xmlns:news="http://www.google.com/schemas/sitemap-news/0.9"
  xmlns:xhtml="http://www.w3.org/1999/xhtml"
  xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
>
  <!-- ... -->
</urlset>
```

> Note: The xmlns:video="..." line is missing from the XML.

Reviewer 2: ```suggestion
```
I don't think we need to call this out like this.

But we *could* consider a diff in the code sample maybe? We could have a `-` line that shows `xmlns:video` ? If so, can we show what the line would look like (in full, no `...` but the actual line Astro creates) if you hadn't excluded it? That could be a good example because then people can see what is cluttering up their XML that they didn't need!
